runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -39,16 +41,23 @@
     private final CorfuRuntime runtime;
     private final CorfuGuidGenerator guidGenerator;
 
-    public CorfuQueue(CorfuRuntime runtime, String streamName) {
+    public CorfuQueue(CorfuRuntime runtime, String streamName, ISerializer serializer,
+                      CorfuTable.IndexRegistry<Long, E> indices) {
         this.runtime = runtime;
         corfuTable = runtime.getObjectsView().build()
                 .setTypeToken(new TypeToken<CorfuTable<Long, E>>() {})
                 .setStreamName(streamName)
-                .setArguments(CorfuTable.IndexRegistry.empty(), new LinkedHashMap<Long, E>())
+                .setArguments(indices, new LinkedHashMap<Long, E>())
+                .setSerializer(serializer)
                 .open();
         guidGenerator = CorfuGuidGenerator.getInstance(runtime);
     }
 
+    public CorfuQueue(CorfuRuntime runtime, String streamName) {
+        this(runtime, streamName, Serializers.JSON, CorfuTable.IndexRegistry.empty());
```

Comment:
Why it's JSON not CORFU?

Reply:
I am just retaining the default serializer which is JSON if not specified.